src/graphics/tile.rs

```diff
@@ -1,36 +1,165 @@
 use crate::graphics::color::Color::LightGray;
+use crate::graphics::color::Color::LightCoral;
 use crate::graphics::font::GLChar;
 use crate::graphics::rectangle::RectVertexAttr;
 
+
+// One line of the help menu
+pub struct HelpItem {
+    effect: String,
```

Comment:
Ok, I think that this struct should not be named `HelpItem` as there is nothing about it which is limited to the Help menu. 

Instead, I would
- name it `StringRect` (suggestion, maybe there is something better ?)
- move it to another file `string_rect.rs`, since other menus will be able to use it

Reply:
I'm sorry about the misunderstanding, in my head a `HelpItem` class would be something like

```rust
/// One line of the help menu
struct HelpItem {
    command: String,
    key: String
}

/// All the lines of the help menu
struct HelpMenuModel {
    items: Vec<HelpItem>
}
```

So it's an element of the `MODEL` (vs the `VIEW`).

It's usually considered a good practice to separate the `model` from the `view`, with the general rules that the `model` does not know anything about the `view` (in our case, the fact that we use `Rect`) and the `view` instead knows about the public functions of the `model`.

I can explain in more details if necessary

Reply:
> Ok, I think that this struct should not be named `HelpItem` as there is nothing about it which is limited to the Help menu.
> 
> Instead, I would
> 
>     * name it `StringRect` (suggestion, maybe there is something better ?)
> 
>     * move it to another file `string_rect.rs`, since other menus will be able to use it

Yes indeed I had this in mind and I wanted to checkout with you at witch level it should be abstracted

For your second comment I think I understand. But does the VIEW posses the MODEL as field or are they both possessed by a higher class (like HUDManager ), or even are in two different classes ?

Reply:
Yes, it's fine that the view possesses the model. 

When the view´ internal logic becomes more complex, the generally it becomes "nasty" to put the model in the view and there is a design pattern to solve this problem: MVMV (Model, View Model, View).

But in this case it's really fine that the view owns the model